libp2p/Common.h

```diff
@@ -267,6 +267,16 @@ inline std::ostream& operator<<(std::ostream& _strm, NodeID const& _id)
     return _strm;
 }
 
+inline boost::log::formatting_ostream& operator<<(
+    boost::log::formatting_ostream& _strm, PeerSessionInfo const& _peerSessionInfo)
```

Comment:
> 
> 
> Better to add a version for non-const `PeerSessionInfo&`, too

Why do we need this? I thought that non-const PeerInfo would be converted to const when calling the operator, which is acceptable since we don't need to modify the PeerInfo when logging it?

Reply:
Apparently generic `template<class T> operator<<(formatting_ostream& _strm, T& _value)` is a better fit during overload resolution than implicitly converting to const reference
https://github.com/ethereum/aleth/blob/f5db298d76aa270b534bde15ff9cb36db9fdfb52/libdevcore/Log.h#L136-L138

I found no better solution for this than defining second versions for non-const reference.

Maybe check that this is the case for `PeerSessionInfo`.